lib/components/molecules/Search/Search.js

```diff
@@ -0,0 +1,110 @@
+// @flow
+/**
+ *
+ * Search
+ *
+ */
+import React, { useState, Node, useCallback } from 'react';
+import styled from 'styled-components';
+import classnames from 'classnames';
+
+import { useSpeechRecognition } from '../../../core/hooks';
+import Typeahead from '../Typeahead';
+import Button from '../../atoms/Button';
+import styles from './Search.style';
+import type { Props } from './types';
+
+const Search = ({ className, inputProps, dataLists, getSelectedResultValue }: Props): Node => {
+  const [searchString, setSearchString] = useState(inputProps.value || '');
+  const [searchLists, setSearchList] = useState();
+
+  const onSearchClick = () => {
+    getSelectedResultValue(searchString);
+  };
+
+  const customFilter = useCallback(
+    inputText => {
+      const lists = dataLists.filter(name => {
+        return name.toLowerCase().indexOf(inputText.toLowerCase()) !== -1;
+      });
+      setSearchList(lists);
+    },
+    [dataLists, setSearchList]
+  );
+
+  const onSpeechRecognitionResults = result => {
+    setSearchString(result);
+    customFilter(result);
+  };
+
+  const handleSpeechEnd = () => {};
+
+  const { isListening, onSpeechListenHandler, browserSupported } = useSpeechRecognition({
+    onSpeechRecognitionResults,
+    handleSpeechEnd,
+  });
+
+  const onSpeechClick = () => onSpeechListenHandler();
+
+  return (
+    <div className={classnames(styles.default, className)}>
+      <div className="search-box">
+        <Typeahead
+          className={classnames('search-txt', inputProps.className)}
+          id="typeahead"
```

Comment:
There should be a way to set the ID from outside. This can be a fallback default value